### ISSUE TITLE
[Bug] Routes which are overlapping and have different depth are not found

### ISSUE THREAD

**Issue by nichmor** [sanic-org/sanic-router#14]
**Describe the bug**
Routes which are overlapping and have different depth are not found.

For example, if you describe your paths as :
`/foo/<foo_id>/bars_ids`
`/foo/<foo_id>/bars_ids/<bar_id>/settings`

you will get not found error while requesting /foo/123/bars_ids.

This cause is hidden in `optimize()` function,  when `Line("...", 0, offset=-1, render=False))` are rendered, 
offset is not applied  to structures  which have same indentation but are situated below.

**Code snippet**
```
def find_route(path, router, basket, extra):
    *omitted_part*
                    if parts[2] == "bars_ids":
                        if num > 3:
                            basket[3] = parts[3]
                            if num == 5:
                                if parts[4] == "settings":
                                    .....
                              **this block should have same identation as if num>3**           
                            try:
                                basket['__params__']['foo_id'] = str(basket[1])
```

**Expected behavior**
Both routes should be accessed as expected

**Environment :**
 - OS: MacOS 10.15.2 (19C57)
 - python: Python v3.7.1:260ec2c36a
 - sanic-router== 0.4.1

PR: [#13 ](https://github.com/sanic-org/sanic-router/pull/13)

**Comment by ahopkins** [comment]
Thanks, I'll look into this.

**Comment by ahopkins** [comment]
Fixed by #13

**Comment by ahopkins** [comment]
Packaging and getting a release out
